query/sources/common/src/main/java/org/codice/ddf/admin/sources/utils/SourceUtilCommons.java

```diff
@@ -191,6 +191,23 @@ public Report saveSource(PidField pid, Map<String, Object> serviceProps, String
         return saveSourceReport;
     }
 
+    public Report saveSource(Map<String, Object> serviceProps, String factoryPid) {
+        ReportImpl saveSourceReport = new ReportImpl();
+        if (serviceCommons.createManagedService(serviceProps, factoryPid)
+                .containsErrorMsgs()) {
+            saveSourceReport.addResultMessage(failedPersistError());
+        }
+        return saveSourceReport;
+    }
+
+    public Report updateSource(PidField pid, Map<String, Object> serviceProps) {
+        ReportImpl updateSourceReport = new ReportImpl();
+        if (StringUtils.isNotEmpty(pid.getValue())) {
```

Comment:
This check will likely not be needed anymore since we are now using the PID inside the config for Updating sources, and it is marked as required. Validation will now catch this before we ever hit this.